tests/pfquickcheck/pflang.lua

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env luajit
+-- -*- lua -*-
+-- This module generates (a subset of) pflang, libpcap's filter language
+
+-- Convention: initial uppercase letter => generates pflang expression
+-- initial lowercase letter => aux helper
+
+module(..., package.seeall)
+local choose = require("pf.utils").choose
+local set = require("pf.utils").set
+
+local function Empty() return { "" } end
+
+-- Given something like { 'host', '127.0.0.1' }, make it sometimes
+-- start with src or dst. This should only be called on expressions
+-- which can start with src or dst!
+local function srcdstify(expr)
```

Comment:
actually (and apologies about github's friggin notifications) better to pass the base operator (like "net") and the argument as separate arguments, and build the result without mutating the incoming expr.